tests/icub-model-test.cpp

```diff
@@ -328,15 +406,44 @@ bool checkFTSensorsAreCorrectlyOriented(iDynTree::KinDynComputations & comp)
                        && checkFTSensorIsCorrectlyOriented(comp, rootLink_R_foot_sensorFrameExpected_plus, "r_foot_ft_sensor");
 
     bool ok = checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameExpected, "l_arm_ft_sensor");
-    ok = ok && checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameExpected, "r_arm_ft_sensor");
-    ok = ok && checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameExpected, "l_leg_ft_sensor");
-    ok = ok && checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameExpected, "r_leg_ft_sensor");
-    ok = ok && (checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameExpected, "l_foot_ft_sensor") || isPlusModel);
-    ok = ok && (checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameExpected, "r_foot_ft_sensor") || isPlusModel);
+    ok = ok & checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameExpected, "r_arm_ft_sensor");
+    ok = ok & checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameExpected, "l_leg_ft_sensor");
+    ok = ok & checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameExpected, "r_leg_ft_sensor");
+    ok = ok & (checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameExpected, "l_foot_ft_sensor") || isPlusModel);
+    ok = ok & (checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameExpected, "r_foot_ft_sensor") || isPlusModel);
 
     return ok;
 }
 
+bool checkFTSensorsAreCorrectlyOrientedV3(iDynTree::KinDynComputations & comp)
+{
+    // The rotation of all the F/T sensors in the iCub robot when in zero position are the same except for the feet and
+    // the left arm
+    iDynTree::Rotation rootLink_R_sensorFrameLeftArmExpected =
+        iDynTree::Rotation(-0.267012, -0.961047, 0.0713662,
+                           -0.960459, 0.271447, 0.0619303,
+                           -0.0788901, -0.0520081, -0.995526);
+    iDynTree::Rotation rootLink_R_sensorFrameExpected =
+        iDynTree::Rotation(-0.267012, 0.961047, 0.0713662,
+                            0.960459, 0.271447, -0.0619303,
+                           -0.0788901, 0.0520081, -0.995526);
+
+    iDynTree::Rotation rootLink_R_sensorFrameExpectedFoot =
+        iDynTree::Rotation(-0.5, 0.866025, 0,
+                           -0.866025, -0.5, 0,
+                            0, 0, 1);
+
+    bool ok = checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameLeftArmExpected, "l_arm_ft_sensor");
+    ok = ok & checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameExpected, "r_arm_ft_sensor");
+    ok = ok & checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameExpected, "l_leg_ft_sensor");
+    ok = ok & checkFTSensorIsCorrectlyOriented(comp, rootLink_R_sensorFrameExpected, "r_leg_ft_sensor");
```

Comment:
This is extremely strange. I would expect the FT frames of the FT in the legs to have the Z direction corresponding to the root link, and then having just an additional rotation in the Z direction due to how the sensor is mounted, as it happens in the feet. As we did not validated the FT frames in the model, this may suggest that the frame orientation is not correct, so for the time being I would avoid putting them in the tests if we do not know if this frames are correct.

Reply:
I agree, moreover now the fts in the legs are disabled, then I remove it.